.github/workflows/conan-build.yml

```diff
@@ -0,0 +1,558 @@
+          pipx ensurepath
+          pipx install conan
+
+      - name: Add pipx bin to PATH
+        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
+
+      - name: Setup Conan
+        run: |
+          conan --version
+          conan profile detect
+
+      - name: Install dependencies
+        run: >
+          conan install . \
+            -c tools.cmake.cmaketoolchain:generator=Ninja \
+            -s compiler=gcc \
+            -s compiler.version=${{ matrix.gcc-version }} \
+            -s compiler.cppstd=${{ matrix.cppstd }} \
+            -s build_type=${{ matrix.build-type }} \
+            --build=missing
+
+      - name: CMake Configure
+        run: |
+          cmake --list-presets
+          cmake --preset ${{ matrix.cmake-preset }}
+
+      - name: Build
+        run: |
+          cmake --build --list-presets
+          cmake --build --preset ${{ matrix.cmake-preset }}
+
+      - name: Test
+        run: |
+          ctest --list-presets
+          ctest --preset ${{ matrix.cmake-preset }}
+
+  test-on-appleclang:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-14]
+        cppstd: ['20']
+        build-type: [Debug, Release]
+        arch: ['x86_64', 'armv8']
+        include:
+          - os: macos-14
+            appleclang-version: 15
+          - build-type: Debug
+            cmake-preset: conan-debug
+          - build-type: Release
+            cmake-preset: conan-release
+
+    name: AppleClang ${{ matrix.appleclang-version }} ${{ matrix.arch }} - C++${{ matrix.cppstd }} - ${{ matrix.build-type }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install CMake
+        uses: lukka/get-cmake@latest
+        with:
+          cmakeVersion: 4.1.0
+          ninjaVersion: latest
+
+      - name: Install Xcode 15.4.0
+        if: matrix.appleclang-version == 15
+        uses: maxim-lobanov/setup-xcode@v1.6.0
+        with:
+          xcode-version: '15.4.0'
+
+      - name: Install system dependencies
+        run: |
+          brew update
+          brew install ninja git python3 conan
+
+      - name: Setup Conan
+        run: |
+          conan --version
+          conan profile detect
+
+      - name: Install dependencies
+        run: >
+          conan install . \
+            -c tools.cmake.cmaketoolchain:generator=Ninja \
+            -s compiler=apple-clang \
+            -s compiler.version=${{ matrix.appleclang-version }} \
+            -s compiler.cppstd=${{ matrix.cppstd }} \
+            -s arch=${{ matrix.arch }} \
+            -s build_type=${{ matrix.build-type }} \
+            --build=missing
+
+      - name: CMake Configure
+        run: |
+          cmake --list-presets
+          cmake --preset ${{ matrix.cmake-preset }}
+
+      - name: Build
+        run: |
+          cmake --build --list-presets
+          cmake --build --preset ${{ matrix.cmake-preset }}
+
+      - name: Test
+        run: |
+          ctest --list-presets
```

Comment:
Do we need to cache conan dependencies, like we do with aui.boot? https://github.com/aui-framework/aui/blob/abaefd9113cee9fa026171624cb7d801cc8c7539/.github/workflows/build.yml#L203-L217